apps/studio/components/interfaces/Database/Triggers/TriggersList/TriggersList.tsx

```diff
@@ -44,20 +43,29 @@ const TriggersList = ({
   const schema = schemas?.find((schema) => schema.name === selectedSchema)
   const isLocked = protectedSchemas.some((s) => s.id === schema?.id)
 
-  const triggers = meta.triggers.list()
+  const {
+    data: triggers,
+    error,
+    isLoading,
+    isError,
+  } = useDatabaseTriggersQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+
   const canCreateTriggers = useCheckPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'triggers')
 
-  if (meta.triggers.isLoading) {
+  if (isLoading) {
     return <GenericSkeletonLoader />
   }
 
-  if (meta.triggers.hasError) {
-    return <AlertError error={meta.triggers.error} subject="Failed to retrieve database triggers" />
+  if (isError) {
+    return <AlertError error={error} subject="Failed to retrieve database triggers" />
   }
 
   return (
     <>
-      {triggers.length == 0 ? (
+      {(triggers ?? []).length == 0 ? (
```

Comment:
Doesn't make a difference here, but I can't unsee 🙈
```suggestion
      {(triggers ?? []).length === 0 ? (
```

Reply:
hahaha no worries - i feel the same now that you pointed it out 😂